impl/src/main/java/io/kroxylicious/testing/kafka/common/Utils.java

```diff
@@ -53,28 +58,54 @@ public static Stream<ServerSocket> preAllocateListeningSockets(int num) {
     }
 
     public static void awaitExpectedBrokerCountInCluster(Map<String, Object> connectionConfig, int timeout, TimeUnit timeUnit, Integer expectedBrokerCount) {
-        try (Admin admin = Admin.create(connectionConfig)) {
-            Awaitility.await()
-                    .pollDelay(Duration.ZERO)
-                    .pollInterval(1, TimeUnit.SECONDS)
-                    .atMost(timeout, timeUnit)
-                    .ignoreExceptions()
-                    .until(() -> {
-                        log.info("describing cluster: {}", connectionConfig.get("bootstrap.servers"));
-                        final Collection<Node> nodes;
-                        try {
-                            nodes = admin.describeCluster().nodes().get(10, TimeUnit.SECONDS);
-                            log.info("got nodes: {}", nodes);
-                            return nodes;
-                        }
-                        catch (InterruptedException | ExecutionException e) {
-                            log.warn("caught: {}", e.getMessage(), e);
-                        }
-                        catch (TimeoutException te) {
-                            log.warn("Kafka timed out describing the the cluster");
-                        }
-                        return Collections.emptyList();
-                    }, Matchers.hasSize(expectedBrokerCount));
+        var knownReady = Collections.synchronizedSet(new HashSet<String>());
+        var toProbe = Collections.synchronizedSet(new HashSet<String>());
+
+        var originalBootstrap = String.valueOf(connectionConfig.get(BOOTSTRAP_SERVERS_CONFIG));
+        toProbe.addAll(Arrays.asList(originalBootstrap.split(",")));
```

Comment:
If we are going to probe each broker independently I would prefer to make that explicit in the api and pass each address separately rather than reversing them back out of the bootstrap string.

Having said that in the spirit of progress over polish, I'd be happy to do that in a separate PR.